tpm2/test/hierarchy_change_auth_test.go

```diff
@@ -0,0 +1,75 @@
+package tpm2test
+
+import (
+	"testing"
+
+	. "github.com/google/go-tpm/tpm2"
+	"github.com/google/go-tpm/tpm2/transport/simulator"
+)
+
+func TestHierarchyChangeAuth(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	authKey := []byte("authkey")
+	newAuthKey := []byte("newAuthKey")
+
+	t.Run("HierarchyChangeAuthOwner", func(t *testing.T) {
+		hca := HierarchyChangeAuth{
+			AuthHandle: TPMRHOwner,
+			NewAuth: TPM2BAuth{
+				Buffer: authKey,
+			},
+		}
+
+		_, err := hca.Execute(thetpm)
+		if err != nil {
+			t.Fatalf("failed HierarchyChangeAuth: %v", err)
+		}
+	})
+
+	t.Run("HierarchyChangeAuthOwnerUnauth", func(t *testing.T) {
+		hca := HierarchyChangeAuth{
+			AuthHandle: TPMRHOwner,
+			NewAuth: TPM2BAuth{
+				Buffer: newAuthKey,
+			},
+		}
+
+		_, err := hca.Execute(thetpm)
+		if err == nil {
+			t.Fatal("failed HierarchyChangeAuthWithoutAuth: Expected HierarchyChangeAuth to fail")
+		}
+	})
+
+	t.Run("HierarchyChangeAuthOwnerAuth", func(t *testing.T) {
+		hca := HierarchyChangeAuth{
+			AuthHandle: AuthHandle{
+				Handle: TPMRHOwner,
+				Auth:   PasswordAuth(authKey),
+			},
+			NewAuth: TPM2BAuth{
+				Buffer: newAuthKey,
+			},
+		}
+
+		_, err := hca.Execute(thetpm)
+		if err != nil {
+			t.Fatalf("failed HierarchyChangeAuthWithAuth: %v", err)
+		}
+	})
+
+	t.Run("Clear", func(t *testing.T) {
```

Comment:
Hmm, what aspect of HierarchyChangeAuth is this subtest testing?

Reply:
You are right, this step is unnecessary for this test.